### PR TITLE
Export Transport

### LINKED ISSUE #309: Support httptrace.ClientTrace for annotating traces
We should provide an implementation that harvests events and annotates the current trace (added in #306)

### PR DESCRIPTION
**Pull request by rakyll**
Then, we can use the Transport struct to set options such as
the one required to fix #309.

### CODE REVIEW

File: plugin/http/httptrace/httptrace.go [hunk 1]
```diff
@@ -25,15 +25,23 @@ import (
...
+// Transport is an http.RoundTripper that traces the outgoing requests.
+//
+// Use NewTransport to create new transports.
+type Transport struct {
+	// Base represents the underlying roundtripper that does the actual requests.
+	// If none is given, http.DefaultTransport is used.
+	//
+	// If base HTTP roundtripper implements CancelRequest,
+	// the returned round tripper will be cancelable.
+	Base    http.RoundTripper
```
Comment (ramonza) [inline plugin/http/httptrace/httptrace.go]: why does Base need to be exposed?

Comment (ramonza) [reply plugin/http/httptrace/httptrace.go]: Oh I see you need to set it explicitly now

**Comment by rakyll** [comment]
> a) Add a mutex to protect Base in t.base()

No, holding a mutex for this is an overkill and costly. We can simply say, mutate Base before using Transport. This is the overall style in the standard library. Fields and methods are not expected to be safe for concurrent use by default in Go.

> It isn't clear to me why we need to export this type in order to provide options on it
Because I am trying to make the zero value useful rather than introducing a new type. We should also export the `formats` field and keep NewTransport around only for convenience.

**Comment by rakyll** [comment]
Can you take another look?

**Pull request merged**